### PR TITLE
Merging main into production

### PR DESCRIPTION
**Pull request by petrungaro**
-Added components:
--Form
--CoinList
--Coin
--News
--Footer

-Added API call into App (for coin data)
-Added API call into Form (to get list of base currencies)
-Added API call into News (to get 10 news articles)

### CODE REVIEW

File: src/News.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    return (
+        <div className="news">
+            <h2>Latest Articles</h2>
+            {news.map((article)=>{
+                return (
+                    <article key={article.pubDate}>
+                        <div className="news-image-container">
+                            <img src={article.image_url 
+                                ? article.image_url
```
Comment (SimonOConnor28) [inline src/News.js:36]: Image failed on one of the articles. maybe you could do a ternary if image doesn't show add this image in it's place

File: src/App.css [hunk 2]
```diff
@@ -1,38 +1,357 @@
...
+}
+
+
+.extra-info {
+  opacity: 0;
+  overflow: hidden;
+  max-height: 0px;
+  position: relative;
+  grid-column: 1 / 7;
+  /* background: #F1F1F1; */
```
Comment (SimonOConnor28) [inline src/App.css]: delete this maybe?

File: src/App.css [hunk 3]
```diff
@@ -1,38 +1,357 @@
...
+  display: flex;
+  flex-direction: column;
+  flex: 1 1 50%;
+}
+
+.details-row {
+  display: flex;
+  justify-content: space-between;
+  gap: 10px;
+  /* align-items: center; */
```
Comment (SimonOConnor28) [inline src/App.css]: delete more commented out code

File: src/App.js [hunk 4]
```diff
@@ -1,23 +1,83 @@
...
+import Form from './Form';
+import CoinList from './CoinList';
+import News from './News';
+import Footer from './Footer';
+
+
 
 function App() {
+
+  /* STATE VARIABLES */
```
Comment (jhshim85) [inline src/App.js:15]: great variables names.

File: src/App.js [hunk 5]
```diff
@@ -1,23 +1,83 @@
...
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+
```
Comment (jhshim85) [inline src/App.js:57]: great structure, but may I suggest the below?
<header>
   <h1>
<main>
   <Form/>
   <CoinList/>
   <News/>

File: src/Coin.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    const athStyle = props.ath > 0 ? {color: 'green'} : {color: 'red'};
+    const atlStyle = props.atl > 0 ? {color: 'green'} : {color: 'red'};
+
+    
+    
+
+    return (
+        <li className="coin">
+            <div className="img-container">
+                <img src={props.image} alt="" />
```
Comment (jhshim85) [inline src/Coin.js]: missing alt=''
could use {props.name}

File: src/CoinList.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            <li className="coin header">
+                <h2 className="coin-name-header">Coin</h2>
+                <h2 className="coin-price">Price</h2>
+                <h2 className="change">24h</h2>
+            </li>
+
+            {filteredCoins.slice(0,props.qtyToShow).map((coin) => {
+                return (
+
+                    <Coin
```
Comment (jhshim85) [inline src/CoinList.js:26]: love how you used ternary and setup your coinlist!

File: src/News.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+import axios from "axios";
+import { useState, useEffect } from "react";
+
+const News = () => {
+
+
+    const [news, setNews] = useState([])
+
+
+    useEffect(()=>{
```
Comment (jhshim85) [inline src/News.js:11]: love that you are using another API for news componenet!

**Review by jhshim85 — COMMENTED** [review]
amazing code!
very neat and easy to access and read!

File: src/Form.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
+import axios from 'axios';
+import {useState, useEffect} from 'react';
+
+
+const Form = (props) => {
+
+    const [vsCurrencyList, setVsCurrencyList] = useState([]);
+
+    useEffect(() => {
```
Comment (adammarostica) [inline src/Form.js:9]: Consider using a debounce hook to reduce the number of API calls you're making: https://usehooks.com/useDebounce/

File: src/App.js (hunk 5 above)
Comment (petrungaro) [reply src/App.js:57]: hi Jay! I don't see any suggestion, can you try rewriting it?

File: src/News.js (hunk 1 above)
Comment (petrungaro) [reply src/News.js:36]: added onError attribute to img tag, which will show the same default image I used in the error handling for when an image link is not provided by the API.
Thanks for catching it!

File: src/Coin.js (hunk 6 above)
Comment (petrungaro) [reply src/Coin.js]: Thanks, added ``

File: src/Form.js (hunk 9 above)
Comment (petrungaro) [reply src/Form.js:9]: Having trouble implementing this without it breaking, will readdress it later. The search is not actually making a new API call fortunately. The API call is made on mount, and when the user changes the currency. 

When a user enters a search term, it's just triggering a regular javascript .filter() on the array of data, so not as intense as making an API call each time, but still not necessary to trigger the filter for each individual keystroke.

Need to get a better understanding of debounce before I reattempt, thanks for pointing this out!

**Pull request merged**